tests/testVariance.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+#
+# LSST Data Management System
+# Copyright 2008-2015 LSST/AURA
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+"""
+Tests for Variance measurement algorithm
+"""
+
+import unittest
+import lsst.utils.tests as utilsTests
+
+import lsst.pex.config as pexConfig
+import lsst.afw.geom as afwGeom
+import lsst.afw.table as afwTable
+import lsst.afw.image as afwImage
+import lsst.afw.detection as afwDetection
+import lsst.meas.base as measBase
+
+try:
+    display
+except NameError:
+    display = False
+
+
+class VarianceTest(unittest.TestCase):
+    def testVariance(self):
+        size = 128 # size of image (pixels)
+        center = afwGeom.Point2D(size//2, size//2) # object center
+        width = 1.0 # PSF width
+        flux = 10.0 # Flux of object
+        variance = 1.0 # Variance value
```

Comment:
Setting the variance to be constant across the image limits your capability for testing. You assert below that the value recovered by your plugin is `1.0`, which is fine insofar as it goes, but it's really only showing that you've successfully managed to read the variance from a single pixel, not that you've managed to select all unmasked pixels within your dilated `Footprint` and find the median variance.

It might be a worthwhile improvement to generate a random (or predictable?) variance image and use that to show your plugin really is doing the calculation it's supposed to.